Open Judge System/OJS.Services/Data/Tests/TestsDataService.cs

```diff
@@ -0,0 +1,19 @@
+﻿namespace OJS.Services.Data.Tests
+{
+    using OJS.Data.Models;
+    using OJS.Data.Repositories.Contracts;
+
+    public class TestsDataService : ITestsDataService
+    {
+        private readonly IEfGenericRepository<Test> tests;
+
+        public TestsDataService(IEfGenericRepository<Test> tests) =>
+            this.tests = tests;
+
+        public void DeleteByProblem(int problemId) =>
+            this.tests.Delete(t => t.ProblemId == problemId);
+
+        public void DeleteByContest(int contestId) =>
+            this.tests.Delete(t => t.Problem.ProblemGroup.ContestId == contestId);
```

Comment:
Similiar comment here, the only case DeleteByProblem would be usefull is when uploading new tests for a problem. This method however seems strange, why would you want to delete all tests for a contest?